sympy/solvers/solveset.py

```diff
@@ -776,6 +778,28 @@ def solveset_complex(f, symbol):
         return result
 
 
+def _solve_complex_hyper(f, symbol):
```

Comment:
You don't need to do all these simplifications, breaking up things into fractions and all, simply rewriting the equation into `exp` should be enough.